libp2p/utils/future.nim

```diff
@@ -31,3 +32,19 @@ proc anyCompleted*[T](futs: seq[Future[T]]): Future[Future[T]] {.async.} =
 
     let index = requests.find(raceFut)
     requests.del(index)
+
+proc raceAndCancelPending*(
+    futs: seq[SomeFuture]
+): Future[void] {.async: (raises: [ValueError, CancelledError]).} =
```

Comment:
`raceAndCancelPending` should return future that finished the race, aka result from `race`, because one might need this future.